core/src/main/java/org/jruby/ir/targets/ConstantLookupSite.java

```diff
@@ -248,7 +248,15 @@ private void bind(Ruby runtime, RubyModule module, IRubyObject constant, MethodH
         MethodHandle fallback = getFallback(module, cachingFallback);
 
         // Test that module is same as before
-        target = guardWithTest(module.getIdTest(), target, fallback);
+        MethodHandles.Lookup lookup = MethodHandles.lookup();
+        MethodHandle testModuleMatch = null;
+        try {
+            testModuleMatch = lookup.findStatic(Bootstrap.class, "testModuleMatch", MethodType.methodType(boolean.class, ThreadContext.class, IRubyObject.class, int.class));
```

Comment:
Same issue here with excessive module creation. I'm sure we can find a way/place to save these handles without polluting things so much that Android doesn't work.